core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,710 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for classification algorithms."""
+
+import collections
+import keyword
+import math
+import StringIO
+import token
+import tokenize
+
+from core.classifiers import base
+from core.classifiers.CodeClassifier import winnowing
+
+import numpy as np
+from sklearn.feature_extraction import text as sklearn_text
+from sklearn import model_selection
+from sklearn import svm
+
+# The string with which all the variable and method names need to be replaced.
+VAR_TOKEN = 'V'
+
+# The string with which will all unkown tokens (tokens which are ignored because
+# they appear rarely in a program) will be replaced.
+UNK_TOKEN = 'UNK'
+
+# T is the model parameter of winnowing algorithm. It is the minimum length of a
+# substring that winnowing should match when comparing two programs.
+# The minimum value of T.
+T_MIN = 3
+# The maximum value of T.
+T_MAX = 11
+
+# K is length of substring for which the hash value is to be generated. Each
+# substring of length K in program will be converted to equivalent hash
+# value (K <= T).
+# Minimum value of K (K <= T). K is model parameter of winnowing algorithm.
+K_MIN = 3
+
+# ID of the token which is a comment.
+COMMENT_TOKEN = 54
```

Comment:
I seem to have confused this with constants used in `token` library. Actually `token.N_TOKENS` is also 53 which I thought is used for `newline` and 54 being used for comment but it is other way around. 53 is used for comment and 54 is used for newline. See this [block](https://github.com/python/cpython/blob/2.7/Lib/tokenize.py#L39) of code.

Actually the library's [docstring](https://github.com/python/cpython/blob/2.7/Lib/tokenize.py#L8) says that it used token type from `token` module, so I didn't check whether they have defined new tokens in `tokenize` module. (Moreover they do `from token import *`  so you never know what is defined in this module and what is defined in other module).

Thanks for checking that out I'll modify the conditions to use `tokenize.NL` and `tokenize.COMMENT` values.